src/Compilers/CSharp/Portable/Symbols/BaseTypeAnalysis.cs

```diff
@@ -29,32 +28,44 @@ internal static bool TypeDependsOn(NamedTypeSymbol depends, NamedTypeSymbol on)
             return result;
         }
 
-        private static void TypeDependsClosure(NamedTypeSymbol type, CSharpCompilation currentCompilation, HashSet<Symbol> partialClosure)
+        private static void TypeDependsClosure(TypeSymbol type, CSharpCompilation currentCompilation, HashSet<Symbol> partialClosure)
         {
             if ((object)type == null)
             {
                 return;
             }
 
-            type = type.OriginalDefinition;
-            if (partialClosure.Add(type))
+            if (type is NamedTypeSymbol namedType)
             {
-                if (type.IsInterface)
+                namedType = namedType.OriginalDefinition;
+                if (partialClosure.Add(namedType))
                 {
-                    foreach (var bt in type.GetDeclaredInterfaces(null))
+                    if (namedType.IsInterface)
                     {
-                        TypeDependsClosure(bt, currentCompilation, partialClosure);
+                        foreach (var bt in namedType.GetDeclaredInterfaces(null))
+                        {
+                            TypeDependsClosure(bt, currentCompilation, partialClosure);
+                        }
                     }
-                }
-                else
-                {
-                    TypeDependsClosure(type.GetDeclaredBaseType(null), currentCompilation, partialClosure);
-                }
+                    else if (namedType.IsExtension)
+                    {
+                        TypeDependsClosure(namedType.GetDeclaredExtensionUnderlyingType(), currentCompilation, partialClosure);
 
-                // containment is interesting only for the current compilation
-                if (currentCompilation != null && type.IsFromCompilation(currentCompilation))
-                {
-                    TypeDependsClosure(type.ContainingType, currentCompilation, partialClosure);
+                        foreach (var bt in namedType.GetDeclaredBaseExtensions())
```

Comment:
>foreach (var bt in namedType.GetDeclaredBaseExtensions()) [](http://example.com/codeflow?start=24&length=57)

Are we testing effect of this?

Reply:
Yes, see `TypeDepends_CircularityViaBaseExtensions`